src/DotVVM.Framework/Controls/HtmlGenericControl.cs

```diff
@@ -194,6 +200,28 @@ private void AddCssClassesToRender(IHtmlWriter writer)
             if (cssClassBindingGroup != null) writer.AddKnockoutDataBind("css", cssClassBindingGroup);
         }
 
+        private void AddStylePropertiesToRender(IHtmlWriter writer)
+        {
+            KnockoutBindingGroup stylePropertiesBindingGroup = null;
+            foreach (var styleProperty in StyleProperties.Properties)
+            {
+                if (stylePropertiesBindingGroup == null) stylePropertiesBindingGroup = new KnockoutBindingGroup();
+                stylePropertiesBindingGroup.Add(styleProperty.GroupMemberName, this, styleProperty);
+
+                try
+                {
+                    var value = GetValue(styleProperty).ToString();
+                    writer.AddStyleAttribute(styleProperty.GroupMemberName, value);
```

Comment:
When the value is null **or empty**, the style property is not set by knockout. The null case is actually consistent with behavior of this, but I think that we could do that without the NullReferenceException ;)

Also note that when the value contains invalid stuff (like a malitious input), the assignment is ignored on client side but can do some harm on the server. I'm not sure if we can do anything about it, but it's worth noting.